packages/sanity/src/core/bundles/components/dialog/BundleForm.tsx

```diff
@@ -24,6 +24,8 @@ export function BundleForm(props: {
 }): JSX.Element {
   const {onChange, onError, value} = props
   const {title, description, icon, hue /*, publishAt*/} = value
+  const {current: action} = useRef(value.slug ? 'edit' : 'create')
```

Comment:
This feels a bit hard to follow, can we pass down the action as a prop instead? 
It feels like it could be mutated from inside this component without really wanting that

Reply:
So that's how I had it - passing `formAction` in from the `BundleDetailsDialog`. However I then thought about scenarios where an initial value was not set, but the action was wrongly set to `edit`, that's why I made a last commit change to infer the action from the initial value passed in. I can add a comment to clarify what is happening and why